tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
```

Comment:
Not intentional no. Just was more focused on nailing the test meaning over the helper implementation.